### PR TITLE
Created a React frontend with Tailwind and TypeScript

### LINKED ISSUE #5: Setup a React front-end
Done when react is set up with:

* development server with automatic code reloading
* typescript
* apollo-client connecting to the server's graphql endpoint

DA already has a react project that uses typescript and apollo-client [here](https://github.com/distributeaid/toolbox/tree/saga/react_ui). It's also set up with auth0, tailwindcss, and a basic layout. The best way to approach this task is probably to migrate that code over to this project.

### PR DESCRIPTION
**Pull request by deammer**
### Changes:
* set up a React app
* added Tailwind
* added Typescript support
* hooked up Apollo for client-side queries
* added an example of how to run queries

### Decisions

* the frontend lives under the `/frontend` directory to make the distinction between backend and frontend clearer. I'm open to feedback!
* running this as a client-side app will allow us to use free hosting (Netlify, etc.)
* Tailwind and CRA don't mesh super well, so there's a bit of a hack in there, but it works 😎 I've documented it in the README

### Look at those pixels!

|Home page|Apollo demo page|
|---|---|
|Screen Shot 2021-02-22 at 8 46 17 PM|Screen Shot 2021-02-23 at 4 59 40 PM|

Resolves #5

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# CSS files output
+*.output.css
```
Comment (bion) [inline frontend/.gitignore:50]: wow this is great and comprehensive, I'll make a note to add a bunch of these to the project base .gitignore

File: README.md [hunk 2]
```diff
@@ -11,8 +11,8 @@
 Install system dependencies:
 
 - node.js v14.15.5
-- yarn v1.22.5
-- postgresql v13.2
+- yarn v1.22.5: `brew install yarn`
+- postgresql v13.2: `brew install postgresql`
```
Comment (bion) [inline README.md:15]: 🙏🏻

File: frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+1. **Install the dependencies CLI**
+
+   ```shell
+   yarn install
+
+   ```
+
+2. **Start the app**
+
+   Get your Gatsby blog set up in a single command:
```
Comment (bion) [inline frontend/README.md]: Gatsby?

File: frontend/src/data/apolloClientInstance.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { ApolloClient, InMemoryCache } from '@apollo/client'
+
+const apolloClientInstance = new ApolloClient({
+  uri: 'http://localhost:3000/graphql',
```
Comment (bion) [inline frontend/src/data/apolloClientInstance.ts]: What's the best way to make this configurable for deployment?

File: frontend/src/pages/ApolloDemo.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+              {JSON.stringify(data, null, 2)}
+            </pre>
+          </>
+        )}
+      </div>
+    </main>
+  )
+}
+
+export default ApolloDemoPage
```
Comment (bion) [inline frontend/src/pages/ApolloDemo.tsx:48]: This demo is so helpful, thank you!

**Review by bion — APPROVED** [review]

File: frontend/README.md (hunk 3 above)
Comment (deammer) [reply frontend/README.md]: 🤦 that's what happens when you copy-paste from nice readmes

File: frontend/src/data/apolloClientInstance.ts (hunk 4 above)
Comment (deammer) [reply frontend/src/data/apolloClientInstance.ts]: I've addressed this with environment variables. See `.env` and the new section in the README.

**Pull request merged**